packages/@lwc/rollup-plugin/src/index.ts

```diff
@@ -38,21 +56,37 @@ function parseQueryParamsForScopedOption(id) {
     };
 }
 
-module.exports = function rollupLwcCompiler(pluginOptions = {}) {
-    const { include, exclude } = pluginOptions;
-    const filter = pluginUtils.createFilter(include, exclude);
+export default function lwc(pluginOptions: RollupLwcOptions = {}): Plugin {
+    const filter = pluginUtils.createFilter(pluginOptions.include, pluginOptions.exclude);
 
-    let customResolvedModules;
-    let customRootDir;
+    let { rootDir, modules = [] } = pluginOptions;
+    const { stylesheetConfig, sourcemap = false, preserveHtmlComments } = pluginOptions;
 
     return {
         name: 'rollup-plugin-lwc-compiler',
 
-        options({ input }) {
-            const { modules: userModules = [], rootDir } = pluginOptions;
+        buildStart({ input }) {
+            if (rootDir === undefined) {
+                if (Array.isArray(input)) {
+                    rootDir = path.dirname(input[0]);
+
+                    if (input.length > 1) {
+                        this.warn(
+                            `The "rootDir" option should be explicitly set when passing an "input" array to rollup. The "rootDir" option is implicitly resolved to ${rootDir}.`
+                        );
+                    }
+                } else {
+                    rootDir = path.dirname(Object.values(input)[0]);
+
+                    this.warn(
+                        `The "rootDir" option should be explicitly set when passing "input" object to rollup. The "rootDir" option is implicitly resolved to ${rootDir}.`
+                    );
+                }
+            } else {
+                rootDir = path.resolve(rootDir);
+            }
```

Comment:
Those 2 questions are related to one another.

In my original PR, I decided to add warnings whenever the rollup plugin is missed configured. Currently, the plugin fails rollup receives an object or an array as [`input`](https://rollupjs.org/guide/en/#input). 

I decided to move this logic from `options` to `buildStart`, because the [`options`](https://rollupjs.org/guide/en/#options) hook is the only plugin that doesn't have access to the plugin context and more specifically `this.warn`.